src/engine.c

```diff
@@ -197,8 +197,12 @@ void engine_run(engine_init_flags *init_flags) {
                     if(e.key.keysym.sym == SDLK_F5) {
                         visual_debugger = !visual_debugger;
                     }
-                    if(!console_window_is_open() && e.key.keysym.sym == SDLK_SPACE) {
-                        debugger_proceed = 1;
+                    if(visual_debugger && !console_window_is_open() && e.key.keysym.sym == SDLK_SPACE) {
+                        dynamic_wait += 20;
+                        static_wait += 20;
+                    } else if(visual_debugger && !console_window_is_open() &&
+                              (e.key.keysym.sym >= SDLK_1 && e.key.keysym.sym <= SDLK_9)) {
+                        debugger_proceed = 1 + e.key.keysym.sym - SDLK_1;
```

Comment:
So this allows using the number keys to step more?

Reply:
Yeah. `1` is also guaranteed to step exactly one dynamic tick, whereas spacebar steps some number of milliseconds.